feeds/wlan-ap/opensync/patches/26-channel-switch-events.patch

```diff
@@ -0,0 +1,174 @@
+Index: opensync-2.0.5.0/interfaces/opensync_stats.proto
+===================================================================
+--- opensync-2.0.5.0.orig/interfaces/opensync_stats.proto
++++ opensync-2.0.5.0/interfaces/opensync_stats.proto
+@@ -678,6 +678,12 @@ message VideoVoiceReport {
+ 	optional uint64 timestamp_ms = 7;
+ }
+ 
++
++enum ChannelSwitchReason {
++        radar_detected                               = 0;
++        high_interference                            = 1;
++    }
++
+ message EventReport {
+     /* Client Association Event */
+     message ClientAssocEvent {
+@@ -791,6 +797,14 @@ message EventReport {
+         optional uint32                 timestamp_ms                        = 6;
+     }
+ 
++    /* Channel Switch Event */
++    message ChannelSwitchEvent {
++       required RadioBandType          band                                = 1;
++       required ChannelSwitchReason    reason                              = 2;
++       required uint32 		       channel  	           	   = 3;
++       optional uint64         	       timestamp_ms 			   = 4;
++    }
++
+     /* Client Session */
+     message ClientSession {
+         required uint64                 session_id                          = 1;
+@@ -877,6 +891,10 @@ message EventReport {
+ 
+     /* Multiple Client Sessions */
+     repeated ClientSession              client_session                      = 1;
++
++    /* Multiple Channel Switch events */
++    repeated ChannelSwitchEvent         channel_switch                      = 2;
++
+     /* Multiple DHCP Transactions */
+     repeated DhcpTransaction            dhcp_transaction                    = 3;
+ }
+Index: opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_events.h
+===================================================================
+--- opensync-2.0.5.0.orig/src/lib/datapipeline/inc/dpp_events.h
++++ opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_events.h
+@@ -40,6 +40,9 @@ typedef enum { CTR_IDLE_TOO_LONG = 0, CT
+ /* proto: SecurityType */
+ typedef enum { SEC_OPEN = 0, SEC_RADIUS, SEC_PSK } sec_type_t;
+ 
++/* proto: CSReason*/
++typedef enum { RADAR_DETECTED = 0, HIGH_INTERFERENCE} cs_reason_t;
++
+ /* proto: ClientAssocEvent */
+ typedef struct {
+ 	char __barrier[46];
+@@ -184,6 +187,22 @@ typedef struct {
+ 	ds_dlist_node_t node;
+ } dpp_event_record_t;
+ 
++/*proto: ChannelSwitchEvent*/
++
++typedef struct {
++       radio_type_t    band;
++       cs_reason_t     reason;
++       uint8_t         freq;
```

Comment:
I think we are trying to send the changed channel to cloud. This should be channel and not frequency